src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/MergeLicenseResourceTransformer.kt

```diff
@@ -0,0 +1,171 @@
+package com.github.jengelman.gradle.plugins.shadow.transformers
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction
+import java.io.OutputStream
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.LinkedHashSet
+import javax.inject.Inject
+import org.apache.tools.zip.ZipEntry
+import org.apache.tools.zip.ZipOutputStream
+import org.gradle.api.file.FileTreeElement
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.util.PatternSet
+
+/**
+ * Generates a license file using the configured license text source.
+ *
+ * A mandatory `SPDX-License-Identifier` is placed in front of the license text to avoid ambiguous
+ * license detection by license-detection-tools.
+ *
+ * License texts found in the files names `META-INF/LICENSE`, `META-INF/LICENSE.txt`,
+ * `META-INF/LICENSE.md`, `LICENSE`, `LICENSE.txt`, `LICENSE.md` are included from the shadow jar
+ * sources. Use the [PatternFilterable][org.gradle.api.tasks.util.PatternFilterable] functions to
+ * specify a different set of files to include, the paths mentioned above are then not considered
+ * unless explicitly included.
+ */
+@Suppress("unused")
+@CacheableTransformer
+public open class MergeLicenseResourceTransformer(
+  objectFactory: ObjectFactory,
+  patternSet: PatternSet,
+) : PatternFilterableResourceTransformer(patternSet) {
+  private val initializePatternSet by lazy {
+    if (patternSet.isEmpty) {
+      includeDefaults()
+    }
+  }
+
+  @get:Internal
+  internal val elements: MutableSet<String> = LinkedHashSet()
+
+  public fun includeDefaults(): MergeLicenseResourceTransformer {
+    patternSet.include(
+      "META-INF/LICENSE",
+      "META-INF/LICENSE.txt",
+      "META-INF/LICENSE.md",
+      "LICENSE",
+      "LICENSE.txt",
+      "LICENSE.md",
```

Comment:
Should we include these non-META-INF paths by default?

Reply:
I'm open for either.
I've been looking into a bunch of dependency jars recently. Most put those `LICENSE*` into `META-INF/`, but quite a few in the "root folder".
There's sadly no standard :(